From d24f2d7269097c3db9990ae2168670ee21720677 Mon Sep 17 00:00:00 2001
From: chenzilin <chenzilin115@gmail.com>
Date: Sat, 29 Aug 2015 18:41:41 +0800
Subject: [PATCH] always use static zlib

---
 src/3rdparty/zlib_dependency.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/3rdparty/zlib_dependency.pri b/src/3rdparty/zlib_dependency.pri
index 0bcb9f9..ca9b384 100644
--- a/src/3rdparty/zlib_dependency.pri
+++ b/src/3rdparty/zlib_dependency.pri
@@ -1,6 +1,6 @@
 # zlib dependency satisfied by bundled 3rd party zlib or system zlib
 contains(QT_CONFIG, system-zlib) {
-    if(unix|mingw):LIBS_PRIVATE += -lz
+    if(unix|mingw):LIBS_PRIVATE += -Wl,-Bstatic -lz -Wl,-Bdynamic
     else {
         isEmpty(ZLIB_LIBS): LIBS += zdll.lib
         else: LIBS += $$ZLIB_LIBS
-- 
1.9.1

