diff --git a/huatai.pro b/huatai.pro
index a27df84..a0b9e2b 100644
--- a/huatai.pro
+++ b/huatai.pro
@@ -13,6 +13,10 @@ CONFIG(release, debug|release) {
         DEFINES += QT_NO_DEBUG_OUTPUT
 }
 
+DEFINES += INITROOT_STARTUP
+LIBS += -linitroot_startup
+CONFIG += qtquickcompiler
+
 # Sources and Headers
 SOURCE_FOLDER = src
 HEADER_FOLDER = src/include
diff --git a/src/main.cpp b/src/main.cpp
index cec9361..60a2cc6 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -15,7 +15,7 @@
 
 #include <ring.h>
 #include <htcarstatus.h>
-
+#include <initroot_startup.h>
 static pid_t pid;
 static bool exit_loop = true;
 
@@ -95,11 +95,12 @@ int server_main(int argc, char *argv[])
 	app.connect(viewer.engine(),&QQmlEngine::quit, &QGuiApplication::quit);
 	viewer.show();
 
-	return app.exec();
+	return startup_end(app.exec);
 }
 
 int main(int argc, char **argv)
 {
+	startup_begin();
 	if (signal(SIGINT, signal_handler) == SIG_ERR ||
 			signal(SIGTERM, signal_handler) == SIG_ERR) {
 		qDebug() << "main process register signal handler fail\n";
