From a69260fca56d8bbacd509b681cbee106a1be51e0 Mon Sep 17 00:00:00 2001
From: chenzilin <chenzilin115@gmail.com>
Date: Fri, 24 Jul 2015 13:58:54 +0800
Subject: [PATCH] enable initfs and qtquickcompiler

---
 src/main.cpp | 5 ++++-
 todi.pro     | 3 +++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 53f44a0..647d213 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -19,6 +19,8 @@
 #include "ringCut.h"
 #include <todi_ui_carstatus.h>
 
+#include <initroot_startup.h>
+
 static pid_t pid;
 static bool exit_loop = true;
 
@@ -70,11 +72,12 @@ int server_main(int argc, char *argv[])
     viewer.show();
     qDebug() << "8.after Show app: " << QTime::currentTime();
 
-    return app.exec();
+    return startup_end(app.exec);
 }
 
 int main(int argc, char **argv)
 {
+    startup_begin();
     qDebug() << "1.enter main process of todi: " << QTime::currentTime();
 
     if (signal(SIGINT, signal_handler) == SIG_ERR ||
diff --git a/todi.pro b/todi.pro
index 154bcf9..26532ed 100644
--- a/todi.pro
+++ b/todi.pro
@@ -9,6 +9,9 @@ TEMPLATE = app
 DEFINES += Todi
 #DEFINES += DEBUG
 DEFINES += TIMER_FREQ
+DEFINES += INITROOT_STARTUP
+LIBS += -linitroot_startup
+CONFIG += qtquickcompiler
 
 # Cross Compile
 cross_compile {
-- 
1.9.1

