From 1ec4fedde409d3ac875a64828bb9d980ef61cbaf Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Thu, 16 Apr 2015 11:13:00 +0800
Subject: [PATCH 2/2] fix meshcvt tool compile fail

---
 tools/meshcvt/meshcvt.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/meshcvt/meshcvt.cpp b/tools/meshcvt/meshcvt.cpp
index 50b6b5a..6cf81f9 100644
--- a/tools/meshcvt/meshcvt.cpp
+++ b/tools/meshcvt/meshcvt.cpp
@@ -42,8 +42,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <QtGui/qvector3d.h>
-#include "qarray.h"
+#include <QVector3D>
+#include <QVector>
 
 static void meshError(const char *filename)
 {
@@ -63,7 +63,7 @@ int main(int argc, char *argv[])
     char buffer[BUFSIZ];
     char *filename;
     char *name;
-    QArray<QVector3D> vertices;
+    QVector<QVector3D> vertices;
 
     // Validate the command-line arguments.
     if (argc < 3) {
-- 
1.9.1

